lms/djangoapps/api_manager/sessions/views.py

```diff
@@ -105,8 +111,16 @@ def post(self, request):
                     # violate our RESTfulness
                     #
                     engine = import_module(settings.SESSION_ENGINE)
-                    new_session = engine.SessionStore()
-                    new_session.create()
+                    if session_id is None:
+                        new_session = engine.SessionStore()
+                        new_session.create()
+                        success_status = status.HTTP_201_CREATED
+                    else:
+                        new_session = engine.SessionStore(session_id)
+                        success_status = status.HTTP_200_OK
+                        if SESSION_KEY in new_session:
+                            # Cannot re-login a user who is already logged in:
+                            response_status = status.HTTP_403_FORBIDDEN
```

Comment:
@bradenmacdonald not 100% sure, but looks like this should be `success_status = status.HTTP_403_FORBIDDEN` - `response_status` value is overwritten by whatever is in success status by the next change, so this would have no effect.